### DISCUSSION TITLE
How to get the matcher from `match`

### DISCUSSION THREAD

**Discussion by Djboy08** [pillarjs/path-to-regexp#294] [Q&A]
I have a case where I give it a array of strings for the matchers. I am using the match function in order to fetch the params if it hits a route that matches the regexp. The issue is I have a object that I want to relate back to based on the route it matches too.

If i simply match the array of strings, I have no way of getting the route it matched on.

i.e:
`/example/:slug/:slug2`
->
`/example/test/test2`

I would only get the input back from the match function:
`/example/test/test2` 

when I also want the route itself,
`/example/:slug/:slug2` so that I can match it against a object. Please i would love insight on this.

**Comment by blakeembrey** [comment]
It's not possible because the input strings are not retained. I would recommend you pass process them individually and use a simple `for` loop to check which one matches, therefore keeping a map of input -> fn -> output.
